### DISCUSSION TITLE
Question about load checkpoint for my own model

### DISCUSSION THREAD

**Discussion by exiawsh** [Lightning-AI/pytorch-lightning#11718] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello, I have defined my Lightning module like this 
![图片](https://user-images.githubusercontent.com/74858581/152343650-475dea27-5a80-49ee-8ecc-fb7a44b4908d.png)
But when I load the checkpoint, I got error, how can I load my own cfg and checkpoint corrcetly?
![图片](https://user-images.githubusercontent.com/74858581/152343784-2c8b4ff0-f2fa-4314-a5d3-d8db57951ad8.png)

**Comment by rohitgr7** [comment]
hey @exiawsh !

two things here:
1. `load_from_checkpoint` is a class method so you need to call it using `TrainingModule.load_from_checkpoint`
2. Since cfg is not part of hyperparameters, you need to call it like this `TrainingModule.load_from_checkpoint(checkpoint_path, cfg=cfg)`

check out the note and examples here: https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#load-from-checkpoint
